### PR TITLE
Journalist can edit their own article

### PR DESCRIPTION
**Pull request by johannaanden**
## Features
- Adds connection between User and Article model
- Adds edit and update actions for the journalist article controller
- Adds edit view
- Adds edit and update routes for the journalist namespace
- Adds roles using enum
- Restricts edit function to author of the article
- Adds unit test
- Refactors every test to reflect the connection between the user and article model

## Pivotal Tracker
- [PT Link](https://www.pivotaltracker.com/story/show/163088265)

### CODE REVIEW

File: app/controllers/journalist/articles_controller.rb [hunk 1]
```diff
@@ -6,14 +6,28 @@ def new
...
+        if @article.update(article_params.merge(category: find_category))
+            redirect_to article_path(@article), notice: 'Updated successfully!'
+        else
+            redirect_to edit_journalist_article_path, notice: 'Every field needs to be filled in!'
+        end
+    end
+
     private
     def article_params
         params.require(:article).permit(:title, :description, :content, :journalist)
```
Comment (Jon1969Edwards) [inline app/controllers/journalist/articles_controller.rb]: Should we add :category to the article_params?

Comment (GergKllai1) [reply app/controllers/journalist/articles_controller.rb]: We have a helper private method to take care of that

**Review by HannaSwDn — APPROVED** [review]

File: config/routes.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@
   root controller: :articles, action: :index
   resources :articles, only: [:index, :show]
   namespace :journalist do
```
Comment (oliverochman) [inline config/routes.rb]: Change the namespace name to `content_manegment`

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: config/routes.rb (hunk 2 above)
Comment (johannaanden) [reply config/routes.rb]: done

**Review by oliverochman — APPROVED** [review]
Good job guys!

**Pull request merged**